rust/connlib/clients/shared/src/eventloop.rs

```diff
@@ -321,6 +324,7 @@ where
 
                 tracing::debug!(resource_id = %offline_resource, "Resource is offline");
 
+                self.tunnel.offline_resource(offline_resource);
```

Comment:
Does it make sense to have a 2nd function here? Couldn't we perfomr `cleanup_connection` now as part of `offline_resource`?

Reply:
`cleanup_connection` is used in other places, this is the only place where we want to consider a "resource offline"

Reply:
So can we call `cleanup_connection` from `offline_resource`?

Reply:
yeah, that works